HIP/hip-435.md

```diff
@@ -0,0 +1,178 @@
+
+As an operator of a mirror node, I want the ability to download sidecar record files containing all the information about records execution that a Hedera node can provide.
+
+## Specification
+
+Introduce a sidecar transaction record protobuf message and a sidecar record file protobuf message to enable tracking of additional information about records execution.
+
+### Record Stream File Changes
+
+A new version (V6) of the record stream file format will be created using the following protobufs:
+
+```
+message RecordFileObject {
+  Transaction transaction = 1;
+  TransactionRecord record = 2;
+}
+```
+
+`Transaction` and `TransactionRecord` are already defined as protobuf messages and are used by the Record Stream algorithm. No changes will be required for these messages.
+
+`RecordFileObjects` are serialized in record stream files.
+
+```
+message RecordStreamFile {
+  // Running Hash of all RecordFileObjects before writing this file.
+  bytes startObjectRunningHash = 1; 
+
+  // List of all the record file objects from that period.
+  repeated RecordFileObject fileObjects = 2; 
+
+  // Running Hash of all RecordFileObjects before closing this file.
+  bytes endObjectRunningHash = 3; 
+
+  // The block number associated with this period.
+  uint64 blockNumber = 4; 
+
+  // List of the hashes of all the Sidecar record files created for the same period.
+  // Allows multiple sidecar files to be linked to this RecordStreamFile
+  repeated bytes SideCarFileHash = 5; 
+}
+```
+
+`RecordStreamFile` is a newly defined protobuf message which will be used to serialise all `RecordFileObject`s that are part of the same period into record stream files. Periods are intervals of `hedera.recordStream.logPeriod` seconds used to group `RecordFileObject`s into record stream files.
+The record signature file that is created for each record stream file signs the hash of the entire corresponding stream file. The list of sidecar file hashes is included in the record stream file. This way mirror nodes or any interested party can download the record stream file and all sidecar files and verify that:
+1. repeated bytes SideCarFileHash is correct; 
+2. the signature file signed the correct hash of the entire record stream file;
+
+```
+message TransactionSideCarRecord {
+    // Consensus timestamp will be the same as the consensus timestamp of the 
+    // transaction the side car is related to. This offers a convenient 
+    // way to match record to sidecar. 
+    Timestamp consensusTimestamp = 1;
+    oneof {
+      repeated ContractStateChange = 2; // Example sidecar type
+      repeated ContractAction = 3; // Example sidecar type
+      // Other future categories
+    }
+}
+```
+
+`TransactionSideCarRecord` is a protobuf message which will be used to create sidecar records complementing `TransactionRecord` and storing additional information about a transaction execution. 
+
+```
+message SideCarFile {
```

Comment:
The `c` in Sidecar should not be capitalized. Need to adjust naming here and everywhere else.

```suggestion
message SidecarFile {
```